apps/backend/src/modules/users/user.service.ts

```diff
@@ -46,27 +82,137 @@ class UserService implements Service {
+			notificationFrequency: user.notificationFrequency,
+			passwordHash: user.passwordHash,
+			passwordSalt: user.passwordSalt,
+			updatedAt: user.updatedAt,
+			userTaskDays: user.userTaskDays,
+		}).toObject();
 	}
 
 	public delete(): ReturnType<Service["delete"]> {
 		return Promise.resolve(true);
 	}
 
-	public async find(id: number): Promise<null | UserEntity> {
-		return await this.userRepository.find(id);
+	public async find(id: number): Promise<null | UserDto> {
+		const userEntity = await this.userRepository.find(id);
+
+		if (!userEntity) {
+			throw new UserError({
+				message: ErrorMessage.REQUESTED_ENTITY_NOT_FOUND,
+				status: HTTPCode.BAD_REQUEST,
+			});
+		}
+
+		const user = userEntity.toNewObject();
+
+		if (!user.avatarFileId) {
+			return userEntity.toObject();
+		}
+
+		const avatarEntity = await this.fileService.find(user.avatarFileId);
+
+		const avatar = avatarEntity?.toObject();
+
+		return UserEntity.initialize({
+			avatarFileId: avatar?.id ?? null,
+			avatarUrl: avatar?.url ?? null,
+			createdAt: user.createdAt,
+			email: user.email,
+			id: userEntity.toObject().id,
+			name: user.name,
+			notificationFrequency: user.notificationFrequency,
+			passwordHash: user.passwordHash,
+			passwordSalt: user.passwordSalt,
+			updatedAt: user.updatedAt,
+			userTaskDays: user.userTaskDays,
+		}).toObject();
 	}
 
 	public async findAll(): Promise<UserGetAllResponseDto> {
 		const items = await this.userRepository.findAll();
 
 		return {
-			items: items.map((item) => item.toObject()),
+			items: await Promise.all(
+				items.map(async (itemEntity) => {
+					const item = itemEntity.toNewObject();
+
+					if (!item.avatarFileId) {
+						return itemEntity.toObject();
+					}
+
+					const avatarEntity = await this.fileService.find(item.avatarFileId);
+
+					const avatar = avatarEntity?.toObject();
+
+					return UserEntity.initialize({
+						avatarFileId: avatar?.id ?? null,
+						avatarUrl: avatar?.url ?? null,
+						createdAt: item.createdAt,
+						email: item.email,
+						id: itemEntity.toObject().id,
+						name: item.name,
+						notificationFrequency: item.notificationFrequency,
+						passwordHash: item.passwordHash,
+						passwordSalt: item.passwordSalt,
+						updatedAt: item.updatedAt,
+						userTaskDays: item.userTaskDays,
+					}).toObject();
+				}),
+			),
 		};
 	}
 
-	public findByEmail(email: string): Promise<null | UserEntity> {
-		return this.userRepository.findByEmail(email);
+	public async findByEmail(email: string): Promise<null | UserEntity> {
+		const userEntity = await this.userRepository.findByEmail(email);
+
+		if (!userEntity) {
+			throw new UserError({
+				message: ErrorMessage.REQUESTED_ENTITY_NOT_FOUND,
+				status: HTTPCode.BAD_REQUEST,
+			});
+		}
+
+		const user = userEntity.toNewObject();
+
+		if (!user.avatarFileId) {
+			return userEntity;
+		}
+
+		const avatarEntity = await this.fileService.find(user.avatarFileId);
+
+		const avatar = avatarEntity?.toObject();
```

Comment:
You should add avatar in repository, using relation and initialize entity there

Reply:
Same for others

Reply:
I already did via relations, however, you said that it makes the queries too complex. Can you please explain in a little more detail so that there is no confusion again?

Reply:
It was too complex for updating the avatar, but not getting the user with avatar. For updating we should use fileService, but for getting something like this:
```
const user = await this.userModel
			.query()
			.findById(userId)
			.withGraphJoined(
				`[${RelationName.USER_DETAILS}.[${RelationName.AVATAR_FILE}]]`,
			)
			.execute();
```

Reply:
Then updateAvatar to leave as it is now? Or change it too, if so how?

Reply:
Also, the question is can I add a type to which I will cast the result of the query (using castTo) since the userDetails model does not have a field through which I can access the file model inside the code? If not, what is the best way to make it so that I can access it?

Reply:
Sure, using `castTo` is okay, we use it in other places as well. So you can add a type